src/Symfony/Component/DependencyInjection/Tests/Fixtures/php/services_almost_circular_private.php

```diff
@@ -572,13 +598,13 @@ protected static function getMailerInline_TransportFactory_AmazonService($contai
      */
     protected static function getManager4Service($container, $lazyLoad = true)
     {
-        $a = new \stdClass();
+        $a = ($container->privates['connection4'] ?? self::getConnection4Service($container));
```

Comment:
Side question: why `$a` is not inlined in the `new \stdClass`?

Reply:
This happens when circular loops are found: instantiating connection4 could create manager4 before we exit this function.